### DISCUSSION TITLE
How to get CallableMockInstance type

### DISCUSSION THREAD

**Discussion by UnrilWever** [vitest-dev/vitest#3768] [Q&A]
![image](https://github.com/vitest-dev/vitest/assets/75024633/0851a4f6-50ef-48e0-bee5-db394067004f)
I have to declare a type for vi.fn,but i didn't found how to get it in the doc

**Comment by jwerre** [comment]
I don't think there is a `CallableMockInstance` type.  I solved this problem with the [`Mock`](https://github.com/vitest-dev/vitest/blob/f4e6e99fa3a81fd7a97fb9b435da367dddba7fa4/packages/spy/src/index.ts#L61) type. For example:

```ts
import { describe, beforeAll, it, expect, vi } from 'vitest';
import type { Mock } from 'vitest';

describe('API', () => {

	beforeAll(() => {
		global.fetch = vi.fn(); // mock fetch
	});

	it('make mock request', async () => {
		const mockResponse = {
			data: {
				_id: 'abc123'
			}
		};

		(fetch as Mock).mockResolvedValue({
			status: 200,
			json: function () {
				return new Promise((resolve) => resolve(mockResponse));
			}
		});

		const result = await makeRequest();
		expect(fetch).toHaveBeenCalled();

	});
});
```

Perhaps someone should fix the documentation?

**Reply by sheremet-va** [reply]
Fixed in https://github.com/vitest-dev/vitest/commit/dbf0e97953009e05c6a7beb319738a724365f4fd

Public docs will be updated in the next release. The update will be available shortly at https://main.vitest.dev
